olive/workflows/run/run.py

```diff
@@ -213,13 +213,42 @@ def set_olive_config_for_aml_system(olive_config: dict):
     AzureMLSystem.olive_config = olive_config
 
 
+def _setup_for_winml():
+    # pylint: disable=protected-access
+    # TODO(olivedev): Remove post ORT release >1.22.0
+    try:
+        import onnxruntime as ort
+        from packaging import version
+
+        if version.parse(ort.__version__) >= version.parse("1.22.0"):
+            ort.winml.register_execution_providers()
+            ort._get_available_providers = ort.get_available_providers
+
+            def get_available_providers_winml():
+                # pylint: disable=protected-access
+                providers = ort._get_available_providers()
+                extra_providers = {ep_device.ep_name for ep_device in ort.get_ep_devices()} - set(providers)
+                return providers + list(extra_providers)
+
+            ort.get_available_providers = get_available_providers_winml
+    except Exception as e:
+        logger.warning("An error occurred during WinML setup: %s", e)
+        logger.debug("Exception details:", exc_info=True)
+
+
 def run(
```

Comment:
it is only limited to run.. but we have custom code that not using run

Reply:
so maybe replace all of `get_available_providers`

Reply:
> it is only limited to run.. but we have custom code that not using run

What *custom code*? And, what other code paths are you executing. `run` is the entry point for all config related runs. How/why are you bypassing this call?